Mathlib/RingTheory/Noetherian.lean

```diff
@@ -369,47 +391,80 @@ variable {R M P : Type _} {N : Type w} [Ring R] [AddCommGroup M] [Module R M] [A
   [Module R N] [AddCommGroup P] [Module R P]
 
 theorem finite_of_linearIndependent [Nontrivial R] [IsNoetherian R M] {s : Set M}
-    (hs : LinearIndependent R (coe : s → M)) : s.Finite := by
+    (hs : LinearIndependent R ((↑) : s → M)) : s.Finite := by
   refine'
     by_contradiction fun hf =>
       (RelEmbedding.wellFounded_iff_no_descending_seq.1 (wellFounded_submodule_gt R M)).elim' _
   have f : ℕ ↪ s := Set.Infinite.natEmbedding s hf
-  have : ∀ n, coe ∘ f '' { m | m ≤ n } ⊆ s :=
-    by
-    rintro n x ⟨y, hy₁, rfl⟩
+  have : ∀ n, (↑) ∘ f '' { m | m ≤ n } ⊆ s := by
+    rintro n x ⟨y, _, rfl⟩
     exact (f y).2
-  have : ∀ a b : ℕ, a ≤ b ↔ span R (coe ∘ f '' { m | m ≤ a }) ≤ span R (coe ∘ f '' { m | m ≤ b }) :=
-    by
+  let coe' : s → M := (↑)
+  have : ∀ a b : ℕ, a ≤ b ↔ 
+    span R (coe' ∘ f '' { m | m ≤ a }) ≤ span R ((↑) ∘ f '' { m | m ≤ b }) := by
     intro a b
     rw [span_le_span_iff hs (this a) (this b),
-      Set.image_subset_image_iff (subtype.coe_injective.comp f.injective), Set.subset_def]
+      Set.image_subset_image_iff (Subtype.coe_injective.comp f.injective), Set.subset_def]
     exact ⟨fun hab x (hxa : x ≤ a) => le_trans hxa hab, fun hx => hx a (le_refl a)⟩
   exact
-    ⟨⟨fun n => span R (coe ∘ f '' { m | m ≤ n }), fun x y => by
-        simp (config := { contextual := true }) [le_antisymm_iff, (this _ _).symm]⟩,
-      by dsimp [GT.gt] <;> simp only [lt_iff_le_not_le, (this _ _).symm] <;> tauto⟩
+    ⟨⟨fun n => span R (coe' ∘ f '' { m | m ≤ n }), fun x y => by
+        rw [le_antisymm_iff, (this x y).symm, (this y x).symm, ←le_antisymm_iff, imp_self]
+        trivial⟩,
+      by dsimp [GT.gt]; simp only [lt_iff_le_not_le, (this _ _).symm]; tauto⟩
 #align finite_of_linear_independent finite_of_linearIndependent
 
-/-- If the first and final modules in a short exact sequence are noetherian,
-  then the middle module is also noetherian. -/
+/-- If the first and final modules in a short exact sequence are Noetherian,
+  then the middle module is also Noetherian. -/
+/- Porting note: etaExperiment helps with SemilinearMapClass instance and coercion 
```

Comment:
Please include the original output of mathport in the porting note. I think we should remind ourselves that this went from ~5 lines to ~30 lines.